addons/website/models/ir_rule.py

```diff
@@ -7,8 +7,15 @@ class IrRule(models.Model):
 
     @api.model
     def _eval_context(self):
+        is_frontend = self.env['website']._get_request_website()
         res = super(IrRule, self)._eval_context()
-        res['website_id'] = self.env['website'].get_current_website().id
+        # TODO (review): never been used -> res['website_id'] = self.env['website'].get_current_website().id
+
+        # TODO (review): get_current_website(falback=False) still always return a website in backend (if domain set & match)..
+        # This is why is_frontned is needed or in backend we will also see the default website's company items
+        res['website'] = is_frontend and self.env['website'].get_current_website() or self.env['website']
```

Comment:
is_frontend cannot be done in get_current_Website instead? ?

Reply:
No, `get_current_website` even with `falback=False` still return a website in backend (the one which match the backend URL). I updated the comment to explain that